sdk/servicebus/azure-servicebus/azure/servicebus/management/_management_client.py

```diff
@@ -251,73 +247,610 @@ def update_queue(self, queue_description, **kwargs):
+        if not entry.content:
+            raise ResourceNotFoundError(
+                "Rule('Topic: {}, Subscription: {}, Rule {}') does not exist".format(
+                    subscription_name, topic_name, rule_name))
+        rule_description = RuleDescription._from_internal_entity(entry.content.rule_description)
+        return rule_description
+
+    def create_rule(self, topic, subscription, rule, **kwargs):
+        # type: (Union[str, TopicDescription], Union[str, SubscriptionDescription], Union[str, RuleDescription], Any) -> RuleDescription  # pylint:disable=line-too-long
+        """Create a subscription of a topic.
+
+        :param Union[str, TopicDescription] topic: The topic that will own the to-be-created subscription rule.
+        :param Union[str, SubscriptionDescription] subscription: The subscription that will own the to-be-created rule.
+        :param Union[str, ~azure.servicebus.management.RuleDescription] rule: The rule name or a
+        `RuleDescription` instance. When it's a str, it will be the name of the created rule.
+         Other properties of the created rule will have default values decided by the ServiceBus
+        :rtype: ~azure.servicebus.management.RuleDescription
         """
+        try:
+            topic_name = topic.name  # type: ignore
+        except AttributeError:
+            topic_name = topic
+        try:
+            subscription_name = subscription.name  # type: ignore
+        except AttributeError:
+            subscription_name = subscription
+        try:
+            rule_name = rule.name  # type: ignore
+            to_create = rule._to_internal_entity()  # type: ignore  # pylint:disable=protected-access
+        except AttributeError:
+            rule_name = rule
+            to_create = InternalRuleDescription()  # Use an empty queue description.
 
-        result = []  # type: List[QueueRuntimeInfo]
-        internal_queues = self._list_queues(**kwargs)
-        for queue_name, internal_queue in internal_queues:
-            runtime_info = QueueRuntimeInfo._from_internal_entity(internal_queue)  # pylint:disable=protected-access
-            runtime_info.queue_name = queue_name
-            result.append(runtime_info)
+        create_entity_body = CreateRuleBody(
+            content=CreateRuleBodyContent(
+                rule_description=to_create,  # type: ignore
+            )
+        )
+        request_body = create_entity_body.serialize(is_xml=True)
+        with _handle_response_error():
+            entry_ele = self._impl.rule.put(
+                topic_name,
+                subscription_name,  # type: ignore
+                rule_name,
+                request_body, api_version=constants.API_VERSION, **kwargs)
+        entry = RuleDescriptionEntry.deserialize(entry_ele)
+        result = entry.content.rule_description
         return result
+
+    def update_rule(self, topic, subscription, rule_description, **kwargs):
```

Comment:
Same comment re parameter name `rule` to match the `create` API.

Reply:
updated